lib/sycamore/sycamore/tests/manual/pdf2image.py

```diff
@@ -0,0 +1,41 @@
+import sys
+import logging
+
+from sycamore.utils.pdf import PdfToImageFiles
+
+
+def main(args=None) -> int:
+    if args is None:
+        args = sys.argv[1:]
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(levelname).1s %(asctime)s %(message)s",
+        datefmt="%Y%m%d_%H%M%S",
+    )
+    fn = args.pop(0)
+    full(fn)
+    bail(fn)
+    return 0
+
+
+def full(fn: str) -> None:
+    with PdfToImageFiles(pdf_path=fn, file_dir="/tmp") as gen:
```

Comment:
write the files into a TemporaryDirectory, and verify that everything is cleaned up after the iterator exits but before the tempdir does.